File: crates/opensymphony-gateway-schema/src/planning.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

use super::version::SchemaVersion;

/// Planning session artifact exposed by the gateway.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct PlanningArtifact {
    pub schema_version: SchemaVersion,
    pub artifact_id: String,
    pub session_id: String,
    pub kind: PlanningArtifactKind,
    pub title: String,
    pub content: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub generated_by: Option<String>,
    pub approved: bool,
    pub published_to_tracker: bool,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum PlanningArtifactKind {
    MilestoneDraft,
    IssueDraft,
    SubIssueDraft,
    DependencyMap,
    AcceptanceCriteria,
    VerificationPlan,
    ResearchSummary,
    CodebaseAnalysis,
}

/// Planning session summary for listing.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct PlanningSessionSummary {
    pub schema_version: SchemaVersion,
    pub session_id: String,
    pub project_id: String,
    pub title: String,
    pub status: PlanningSessionStatus,
    pub artifact_count: u32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum PlanningSessionStatus {
    Draft,
    InReview,
    Approved,
    Published,
    Archived,
}

File: packages/gateway-schema/src/planning.ts
import type { SchemaVersion } from "./version.js";

export type PlanningArtifactKind =
  | "intake"
  | "requirements"
  | "milestone_draft"
  | "issue_draft"
  | "sub_issue_draft"
  | "dependency_map"
  | "acceptance_criteria"
  | "verification_plan"
  | "research_summary"
  | "codebase_analysis";

/** Planning session artifact. */
export interface PlanningArtifact {
  schema_version: SchemaVersion;
  artifact_id: string;
  session_id: string;
  kind: PlanningArtifactKind;
  title: string;
  content: string;
  created_at: string;
  updated_at: string;
  generated_by?: string;
  approved: boolean;
  published_to_tracker: boolean;
}

export type PlanningSessionStatus =
  | "draft"
  | "in_review"
  | "approved"
  | "published"
  | "archived";

/** Planning session summary for listing. */
export interface PlanningSessionSummary {
  schema_version: SchemaVersion;
  session_id: string;
  project_id: string;
  title: string;
  status: PlanningSessionStatus;
  artifact_count: number;
  created_at: string;
  updated_at: string;
}
